sema4ai-http-helper/src/sema4ai_http/__init__.py

```diff
@@ -14,60 +16,169 @@
+        protocol: int | None,
+        enable_legacy_server_connect: bool | None = None,
+    ) -> ssl.SSLContext:
+        import truststore
 
+        ssl_context = truststore.SSLContext(protocol or ssl.PROTOCOL_TLS_CLIENT)
 
-def build_ssl_context(
-    protocol: int | None = None, *, enable_legacy_server_connect: bool | None = None
-) -> "ssl.SSLContext":
-    """Returns a Truststore SSL context with OP_LEGACY_SERVER_CONNECT flag enabled"""
-    import os
-    import ssl
+        self._configure_ssl_verification(ssl_context)
 
-    import truststore
+        if enable_legacy_server_connect:
+            self._configure_tls_renegociation(ssl_context)
 
-    # Ignoring typing because trustore has bad typing, it should be int|None, although it says only int.
-    ctx = truststore.SSLContext(protocol)  # type: ignore
-    if enable_legacy_server_connect is None:
-        # i.e.: False if RC_TLS_LEGACY_RENEGOTIATION_ALLOWED is not set
-        env_value = os.getenv("SEMA4AI_TLS_LEGACY_RENEGOTIATION_ALLOWED", "").lower()
-        if not env_value:
-            # Backward compatibility with Robocorp TLS env var
-            env_value = os.getenv("RC_TLS_LEGACY_RENEGOTIATION_ALLOWED", "").lower()
+        return ssl_context
 
-        enable_legacy_server_connect = env_value in ("1", "true")
+    def _configure_ssl_verification(self, ssl_context: ssl.SSLContext) -> None:
+        disable_ssl = self.config.get("disable-ssl", False)
+        ssl_context.verify_mode = ssl.CERT_NONE if disable_ssl else ssl.CERT_REQUIRED
 
-    if enable_legacy_server_connect:
+    def _configure_tls_renegociation(self, ssl_context: ssl.SSLContext) -> None:
         if sys.version_info < (3, 12):
             _SSL_LEGACY_SERVER_CONNECT = 0x4
         else:
             _SSL_LEGACY_SERVER_CONNECT = ssl.OP_LEGACY_SERVER_CONNECT
-        ctx.options |= _SSL_LEGACY_SERVER_CONNECT
-    return ctx
+        ssl_context.options |= _SSL_LEGACY_SERVER_CONNECT
+
+
+class NetworkConfig:
+    def __init__(self) -> None:
+        self.connection_pool = self._build_connection_pool()
+
+    @staticmethod
+    def _get_network_settings_path() -> Path:
+        if sys.platform == "win32":
+            localappdata = os.environ.get("LOCALAPPDATA")
+            if not localappdata:
+                raise RuntimeError("Error. LOCALAPPDATA not defined in environment!")
+            sema4_home = Path(localappdata) / "sema4ai"
+        else:
+            # Linux/Mac
+            sema4_home = Path("~/.sema4ai").expanduser()
+
+        return sema4_home / "network-settings.yaml"
+
+    def _get_current_config_profile(self) -> dict:
+        config_file = self._get_network_settings_path()
+
+        if config_file.exists():
+            import yaml
+
+            config = yaml.safe_load(config_file.read_text())
+            current_profile = config.get("current-profile", "")
+            return next(
+                (p for p in config.get("profiles", []) if p["name"] == current_profile),
```

Comment:
If the user does a bad thing and `p['name']` isn't there, this will raise a keyerror (so, either log and ignore or raise a better error message).